src/dotnet/Microsoft.FxLister/Services/PackageAnalyzer.cs

```diff
@@ -0,0 +1,91 @@
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace Microsoft.FxLister.Services;
+
+public class PackageAnalyzer
+{
+    private readonly ILogger _logger;
+    
+    public PackageAnalyzer()
+    {
+        _logger = NullLogger.Instance;
+    }
+    
+    public async Task<List<string>> DiscoverAzurePackagesAsync()
+    {
+        var azurePackages = new List<string>();
+        
+        try
+        {
+            // Set up NuGet configuration
+            var settings = Settings.LoadDefaultSettings(null);
+            var sourceRepositoryProvider = new SourceRepositoryProvider(settings, Repository.Provider.GetCoreV3());
+            
+            // Get the official NuGet.org repository
+            var nugetOrgSource = sourceRepositoryProvider.GetRepositories()
+                .FirstOrDefault(r => r.PackageSource.Source.Contains("nuget.org"));
+            
+            if (nugetOrgSource == null)
+            {
+                // Fallback to creating nuget.org repository manually
+                var packageSource = new PackageSource("https://api.nuget.org/v3/index.json");
+                nugetOrgSource = new SourceRepository(packageSource, Repository.Provider.GetCoreV3());
+            }
+            
+            var searchResource = await nugetOrgSource.GetResourceAsync<PackageSearchResource>();
+            
+            // Search for Azure packages
+            var searchFilter = new SearchFilter(includePrerelease: false)
+            {
+                SupportedFrameworks = new[] { "net8.0", "netstandard2.0", "netstandard2.1" }
+            };
+            
+            int skip = 0;
+            int take = 20; // Limit to 20 packages for initial testing
```

Comment:
I don't like hardcoding this number here. Let's have it as a command line option